tests/system/src/test/java/test/javafx/stage/SystemMenuBarTest.java

```diff
@@ -108,4 +113,61 @@ public MenuBar createFailingMenuBar() {
 
         return menuBar;
     }
+
+    @Test
+    public void testMemoryLeak() throws InterruptedException {
+        Util.runAndWait(() -> {
+            Thread.currentThread().setUncaughtExceptionHandler((t,e) -> {
+                e.printStackTrace();
+                failed.set(true);
+            });
+            createMenuBarWithItemsStage();
+        });
+        memoryLatch.await();
+        assertFalse(failed.get());
+    }
+
+    public void createMenuBarWithItemsStage() {
+        final ArrayList<WeakReference<MenuItem>> uncollectedMenuItems = new ArrayList<>();
+
+        Stage stage = new Stage();
+        VBox root = new VBox();
+        final MenuBar menuBar = new MenuBar();
+        final Menu menu = new Menu("MyMenu");
+        menuBar.getMenus().add(menu);
+        menuBar.setUseSystemMenuBar(true);
+        root.getChildren().add(menuBar);
+
+        Scene scene = new Scene(root);
+        stage.setScene(scene);
+        stage.show();
+        stage.requestFocus();
+        Thread t = new Thread(){
```

Comment:
Question: Since you do all the work in a `Platform.runLater`, is this thread needed?

Reply:
That's a good question. I do this because we start the test on the FX Thread, but I want to add/remove the menuItems one by one, giving time for a `pulse` in between them. If we do the 50 runs inside a single Runnable, without the possibility for the FX Thread to do other work, the result can be different compared to what we do when doing this in small runnables. 
But I might be missing something here?

Reply:
Oh, I see that now. I had missed that you were calling this method from within a `Util::runAndWait`. Never mind.